presto-raptor/src/test/java/com/facebook/presto/raptor/metadata/TestUpgradeMetadata.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.raptor.metadata;
+
+import org.skife.jdbi.v2.DBI;
+import org.skife.jdbi.v2.Handle;
+import org.skife.jdbi.v2.IDBI;
+import org.skife.jdbi.v2.exceptions.UnableToCreateStatementException;
+import org.skife.jdbi.v2.sqlobject.SqlUpdate;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.util.OptionalLong;
+import java.util.UUID;
+
+import static com.facebook.presto.raptor.metadata.SchemaDaoUtil.createTablesWithRetry;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+/**
+ * Test the correctness of metadata after upgrading
+ * from environment without delta delete to environment with delta delete
+ */
+public class TestUpgradeMetadata
+{
+    IDBI dbi;
+    private MetadataDao dao;
+    private Handle dummyHandle;
+
+    @BeforeMethod
+    public void setup()
+    {
+        dbi = new DBI("jdbc:h2:mem:test" + System.nanoTime());
+        dummyHandle = dbi.open();
+        dao = dbi.onDemand(MetadataDao.class);
+        createTablesForOldEnvironment(dummyHandle.attach(SchemaDao.class), dummyHandle.attach(LegacySchemaDao.class));
+    }
+
+    private void createTablesForOldEnvironment(SchemaDao dao, LegacySchemaDao legacySchemaDao)
+    {
+        dao.createTableDistributions();
+        legacySchemaDao.createTableTablesForTest();
+        dao.createTableColumns();
+        dao.createTableViews();
+        dao.createTableNodes();
+        legacySchemaDao.createTableShardsForTest();
+        dao.createTableShardNodes();
+        dao.createTableExternalBatches();
+        dao.createTableTransactions();
+        dao.createTableCreatedShards();
+        dao.createTableDeletedShards();
+        dao.createTableBuckets();
+        dao.createTableShardOrganizerJobs();
+    }
+
+    @AfterMethod(alwaysRun = true)
+    public void tearDown()
+    {
+        dummyHandle.close();
+    }
+
+    @Test(expectedExceptions = UnableToCreateStatementException.class)
+    public void testTableTablesWithoutUpgrading()
+    {
+        // without perform upgrading operation would cause metadata incorrect
+        long tableId = dao.insertTable("schema1", "table1", true, false, null, 0, false);
+        assertNotNull(tableId);
```

Comment:
No need to create `tableId` and assign/assert on it. The code will never reach this point.